### PR TITLE
Creating Tests and Methods - Test Driven Project 

### PR DESCRIPTION
**Pull request by ouasamine**
## Overview 
In this project, we created a `Solver` class with 3 methods `factorial`, `reverse` and `fizzbuzz` following the TDD flow.

## Workflow
- Set the project files and dependencies
- Created RSpec test for `factorial` and then `factorial`
- Created RSpec test for `reverse` and then `reverse`
- Created RSpec test for `fizzbuzz` and then `fizzbuzz`

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Testing Solver methods' do
```
Comment (1GORDON) [inline spec/solver_spec.rb:4]: Great job team 🤝

- [ ] [Optional] You may also add tests for testing the `raises an exception for negative number`, and `Special case - when input is 0, the result should equal 1`.

**Review by 1GORDON — APPROVED** [review]
# Status: APPROVED 🟢 

Hi Team ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Followed Gitflow ✔️
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ouasamine** [comment]
Thank you @1GORDON
